.github/workflows/build.yml

```
@@ -0,0 +1,307 @@
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+permissions:
+  id-token: write
+  contents: write
+
+jobs:
+  build_maven_parent:
+    runs-on: github-ubuntu-latest-m
+    name: Build Maven Parent
+    outputs:
+      BUILD_NUMBER: ${{ steps.build-maven-action.outputs.BUILD_NUMBER }}
+    permissions:
+      id-token: write
+      contents: write
+    steps:
+      - uses: actions/checkout@v5
+
+      - name: Install dependencies
+        run: |
+          bash .github/workflows/install_dependencies.sh
+
+      - name: Get Repox Credentials from Vault
+        id: secrets
+        uses: SonarSource/vault-action-wrapper@v3
+        with:
+          secrets: |
+            development/artifactory/token/{REPO_OWNER_NAME_DASH}-private-reader username | ARTIFACTORY_USERNAME;
+            development/artifactory/token/{REPO_OWNER_NAME_DASH}-private-reader access_token | ARTIFACTORY_ACCESS_TOKEN;
+
+      - name: Populate Maven settings
+        run: |
+          mkdir ~/.m2 && cp .github/workflows/resources/settings.xml ~/.m2/settings.xml
+
+      - name: Build dotnet project
+        env:
+          # for nuget
+          ARTIFACTORY_USER: ${{ fromJSON(steps.secrets.outputs.vault).ARTIFACTORY_USERNAME }}
+          ARTIFACTORY_PASSWORD: ${{ fromJSON(steps.secrets.outputs.vault).ARTIFACTORY_ACCESS_TOKEN }}
+          # for Maven
+          ARTIFACTORY_ACCESS_TOKEN: ${{ fromJSON(steps.secrets.outputs.vault).ARTIFACTORY_ACCESS_TOKEN }}
+        run: |
+          # Restore the .NET project using Repox as source feed
+          dotnet restore omnisharp-dotnet/SonarLint.OmniSharp.DotNet.Services.sln \
+            --locked-mode \
+            --configfile omnisharp-dotnet/nuget.config
+
+          # Get OmniSharp fork from Repox
+          mvn generate-resources -B -Denable-repo=qa -DskipIts -Pdownload-omnisharp-for-building
+
+          # Build and run .NET unit tests
+          dotnet test omnisharp-dotnet/SonarLint.OmniSharp.DotNet.Services.sln \
+            -c Release \
+            -p:CollectCoverage=true -p:CoverletOutput=TestResults/ -p:CoverletOutputFormat=opencover \
+            --no-restore
+
+      - uses: jdx/mise-action@c37c93293d6b742fc901e1406b8f764f6fb19dac # v2.4.4
+        with:
+          version: 2025.9.12
+
+      - uses: SonarSource/ci-github-actions/build-maven@v1
+        id: build-maven-action
+        with:
+          maven-args: -DskipIts
+          sonar-platform: none
+          artifactory-reader-role: private-reader
+          artifactory-deployer-role: qa-deployer
+          deploy-pull-request: true
+
+      - name: Debug Build Number
+        run: echo "BUILD_NUMBER=${{ steps.build-maven-action.outputs.BUILD_NUMBER }}"
+
+  build_dotnet:
+    runs-on: github-ubuntu-latest-m
+    name: Scan dotnet project on shadow platforms
+    permissions:
+      id-token: write
+      contents: write
+    steps:
+      - uses: actions/checkout@v5
```

It is [recommended by the dotnet scanner](https://docs.sonarsource.com/sonarqube-server/devops-platform-integration/github-integration/adding-analysis-to-github-actions-workflow#configure-build-yml) to disable shallow clones

```suggestion
      - uses: actions/checkout@v5
        with:
          fetch-depth: 0
```